raft.go

```diff
@@ -214,15 +214,13 @@ func (r *Raft) runFollower() {
 				}
 			} else {
 				metrics.IncrCounter([]string{"raft", "transition", "heartbeat_timeout"}, 1)
-				if inConfig(r.configurations.latest, r.localID) {
+				if hasVote(r.configurations.latest, r.localID) {
 					r.logger.Warn("heartbeat timeout reached, starting election", "last-leader", lastLeader)
 					r.setState(Candidate)
 					return
-				} else {
-					if !didWarn {
-						r.logger.Warn("heartbeat timeout reached, not part of stable configuration, not triggering a leader election")
-						didWarn = true
-					}
+				} else if !didWarn {
+					r.logger.Warn("heartbeat timeout reached, not part of stable configuration, not triggering a leader election")
+					didWarn = true
```

Comment:
Is this warning message potentially confusing for this case now? We might be part of a stable configuration just as a non voter yet we'll now output this warning.

The message itself is kinda ambiguous so it might be a technicality since probably no-one understood what it meant anyway, but technically it is no longer necessarily true in the same sense as it was original intended (that this node is not in the committed config at all).

One suggestion would be to simplify this to be accurate and easier to understand: `heartbeat timeout reached, not in stable config or a non-voter so skipping leader election`. Or we could make it conditional and make it clear which one it is depending on whether it's a non-voter or not in the config?

Reply:
I clarified the message without making it conditional as it make the code less complex.